UnityProject/Assets/Scripts/Objects/Climbable.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Climbable : MonoBehaviour, ICheckedInteractable<MouseDrop>
+{
+
+	public float ClimbTime = 1.0f;
+
+
+	public bool WillInteract(MouseDrop interaction, NetworkSide side)
+	{
+		if (!DefaultWillInteract.Default(interaction, side)) return false;
+
+		PlayerSync playerSync;
+		CustomNetTransform netTransform;
+		if(interaction.UsedObject.TryGetComponent(out playerSync))
+		{
+			if (playerSync.IsMoving || playerSync.playerMove.IsBuckled)
+			{
+				return false;
+			}
+
+			// Do a sanity check to make sure someone isn't dropping the shadow from like 9000 tiles away.
+			var mag = (interaction.TargetObject.transform.position - playerSync.ServerPosition).magnitude;
+			if (mag > PlayerScript.interactionDistance)
+			{
+				return false;
+			}
+		}
+		else if(interaction.UsedObject.TryGetComponent(out netTransform)) // Do the same check but for mouse draggable objects this time.
+		{
+			var mag = (interaction.TargetObject.transform.position - playerSync.ServerPosition).magnitude;
+			if (mag > PlayerScript.interactionDistance)
+			{
+				return false;
+			}
+		}
+		else // Not sure what this object is so assume that we can't interact with it at all.
+		{
+			return false;
+		}
+
+		return true;
+	}
+
+	public void ServerPerformInteraction(MouseDrop interaction)
+	{
+		StandardProgressActionConfig cfg =
+			new StandardProgressActionConfig(StandardProgressActionType.Construction, false, false, false);
+		var x = StandardProgressAction.Create(cfg, () =>
+		{
+			PlayerScript playerScript;
+			if (interaction.UsedObject.TryGetComponent(out playerScript))
+			{
+				playerScript.PlayerSync.SetPosition(this.gameObject.WorldPosServer());
```

Comment:
shouldn't this have a small offset in y?

Reply:
wdym? why would we offset this?
If you mean like an offset where it visually appears as if the player is ontop of the crate; I'd rather that we add pixel shifting first instead of messing with the player's position that will probably get set back to an Int due to matrixes and what not.

Reply:
true